addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -214,6 +216,28 @@ def _update_accrual(self):
 
             holiday.write(values)
 
+    def _compute_accrual_plan(self):
+        accrual_plans=self.env['hr.accrual.plan'].search([("company_id","=",self.env.company.id)])
+        for record in self:
+            record.use_accrual_plan=False
+            contract_start=record.employee_id.contract_id.date_start or record.employee_id.contract_ids.date_start
+            # TO DO : level 
+            for accrual_plan in accrual_plans:
```

Comment:
You make the same computation for all the allocations on self.

Could be factorized to improve the performances.